api/src/middleware/rate-limiter.js

```diff
@@ -0,0 +1,46 @@
+const auth = require('../auth');
+const rateLimitService = require('../services/rate-limit');
+const serverUtils = require('../server-utils');
+
+const getKeys = (req, basicAuth) => {
+  const keys = [
+    req.ip,
```

Comment:
Which IP is this?  Specifically, is this the Nginx IP or the actual end user's IP?  

My concern is if it's the Nginx IP, then all traffic will come from the same IP. During something like a training, when a lot of users could fail to log in, we could repeatedly hit the rate-limit, which will affect ALL traffic.

If it's the end user's IP - all good!

Reply:
Hey again! Testing the code, I can see this is correctly the end user's IP - all good!  I'll see myself out the door and resolve the thread - thanks for listening ;)

Reply:
Thank you for taking the time to validate it!